### PR TITLE
Updating project to 2018.2.1f1

### PR DESCRIPTION
**Pull request by StamatisP**
This has fixed mostly everything EXCEPT the lightmaps. Do not enable lightmaps, they break textures.

I have also added some debug logs and a new player controller. Anything else I added, I kinda forgot. Hopefully this is up to your standards!

### CODE REVIEW

File: Assets/Code/Read/SourceBSPLoader.cs [hunk 1]
```diff
@@ -726,104 +740,104 @@ GameObject GenerateFaceObject(int index)
...
 			}
-			
+
 			mf.sharedMesh.RecalculateNormals ();
 			//mf.mesh.RecalculateBounds ();
 
 			faceObject.transform.parent = modelsObject.transform;
 
-			mf.sharedMesh.Optimize ();
+			var o_778_3_636690536766675730 = mf.sharedMesh;
```
Comment (lewa-j) [inline Assets/Code/Read/SourceBSPLoader.cs:792]: WTF?

File: Assets/Code/Read/SourceBSPLoader.cs [hunk 2]
```diff
@@ -833,12 +847,12 @@ void GenerateDispFaceObject(int index, int model)
 			mf.sharedMesh.RecalculateNormals ();
 
 			faceObject.transform.parent = dispObject.transform;
-			mf.sharedMesh.Optimize ();
+			var o_835_3_636690536766835854 = mf.sharedMesh;
```
Comment (lewa-j) [inline Assets/Code/Read/SourceBSPLoader.cs:850]: And here

**Comment by lewa-j** [comment]
Thank you for updating the code. But since there are a lot of style revisions in the same commit, it's hard to keep track of what makes sense. But in fact, I do not care, so it's normal. Can I merge or do you want to finish something else?

**Comment by StamatisP** [comment]
I believe that's everything I wanted to include in. Apologies for the large number of changes, VSCode automatically formats it to make it look a bit nicer.

I added PlayerControllerv2 to mimic the Source-style movement at least a little bit. 

You can see the changes I've made [in here.](https://github.com/lewa-j/Unity-Source-Tools/pull/2/commits/0c4ad46bb8ecad55e4fec918ba2735e65ecdbdef?utf8=%E2%9C%93&diff=unified&w=1) Remember to turn on "hide whitespace"

~~As for the above changes you commented, I have no idea. It was either Unity changing that or VSCode, as I have never touched it.~~ Turns out Unity got rid of the function, and it is now obsolete, and only works in the editor. **https://docs.unity3d.com/ScriptReference/MeshUtility.Optimize.html**

 I didn't mean to include the readme, but it should be fine to keep, right?

File: Assets/Code/Read/SourceBSPLoader.cs [hunk 3]
```diff
@@ -513,98 +526,100 @@ void CreateModelObject(int index)
...
 				{
-					model.AddComponent<MeshCollider>();
+					model.AddComponent<MeshCollider> ();
 				}
 
-				if(!uSrcSettings.Inst.showTriggers)
+				if (!uSrcSettings.Inst.showTriggers)
 				{
-					string materialName = ConvertUtils.GetNullTerminatedString(map.texdataStringDataLump,map.texdataStringTableLump [map.texdataLump[ti.texdata].nameStringTableID]);
+					face = map.facesLump[i];
```
Comment (lewa-j) [inline Assets/Code/Read/SourceBSPLoader.cs]: `i` is not appropriate here. Here you can use `firstFace`.

**Comment by lewa-j** [comment]
>You can see the changes I've made in here. Remember to turn on "hide whitespace"

Oh thanks. I did not know about it.
>Turns out Unity got rid of the function

Well then those lines where it was used can be deleted, it's not so important.
>I didn't mean to include the readme, but it should be fine to keep, right?

Of course. It's good.

**Pull request merged**